### ISSUE TITLE
LocaleProvider should respect LocaleSwitcher

### ISSUE THREAD

**Issue by jontjs** [KnpLabs/DoctrineBehaviors#748]
Since Symfony 6.1, it has been possible to [run a portion of code in a specific locale](https://symfony.com/blog/new-in-symfony-6-1-locale-switcher), and as of Symfony 6.4 to [render emails in a specific locale](https://symfony.com/blog/new-in-symfony-6-4-locale-improvements#render-emails-in-user-locale), but the `LocaleProvider` doesn't seem to respect either of these.

My situation is that I am using the Translatable extension on an e-commerce website, and need to send order-confirmation emails that use the localized names of the entities. When I use either of the new Symfony features, my email template as a whole DOES see use the locale specified. However when I rely on this bundle's `proxyCurrentLocaleTranslation()` method to translate a `translatable` entity's property in the template, the locale used is `en` rather than the one I specified (`en_GB` or `de_DE`). On digging through the code, I think this stems from the `LocaleProvider`'s `provideCurrentLocale()` method

regards,
Jon

**Comment by Gielovic** [comment]
Agree, i would guess replacing this https://github.com/KnpLabs/DoctrineBehaviors/blob/775ccba9bac20baa4add4fef630b66c20211b6da/src/Provider/LocaleProvider.php#L28 with 

        return $this->translator?->getLocale();
would do the trick. 

Recreating the LocaleProvider with this change and wiring the Interface to this new (own) provider did the trick.
